src/ci/ci_server/model.go

```diff
@@ -0,0 +1,23 @@
+package ci_server
+
+import "database/sql"
+
+type CIDB struct {
+	DB *sql.DB
+}
+
+func newCIDB(db *sql.DB) *CIDB {
+	return &CIDB {
+		DB: db,
+	}
+}
+
+func (db *CIDB) AddPushEvent(event *PushEvent) (build_id int64, err error) {
+	addPushEventStmt, err := db.DB.Prepare("select new_push_event(?, ?, ?)")
```

Comment:
这里貌似没问题，益哥说的是不同lexical scope shadow variable的情况。e.g.,
```
func foo() (err error) {
  if err := errors.New("haha"); err != nil {
  }
  // here err is still nil, since the err in if block is in a different lexical scope, so a local variable is created.
}
```
这里两个err是在同一个lexical scope。short variable declaration对err不会创建一个新的变量。测试：
```
package main

import (
	"errors"
	"fmt"
)

func foo() (err error) {
	b, err := bar()
	fmt.Println(b)
	return
}

func bar() (int, error) {
	return 1, errors.New("error")
}

func main() {
	fmt.Println(foo())
}
// prints
// 1
// error
```
go很好的情况是如果
```
func foo(err error) {
  err := errors.New("error") //编译不通过，"no new variables on left side of :="
}
```
shadow的情况确实很需要注意。linter这样的工具就显得尤为重要了。